In [1]:
import autograd as ag

In [2]:
a = ag.Node(3)
b = ag.Node(2)
f = (2 * a + b**a) * b

In [3]:
f.backward()

In [4]:
a.grad, b.grad

(15.090354888959125, 38)

$a=3, b=2$

$f = (2a + b^a)b$

$\frac{\partial f}{\partial a} = 2b + b^{a+1}lnb = 4 + 16ln2 = 15.090$

$\frac{\partial f}{\partial b} = 2a + (a+1)b^a = 6 + 32 = 38$

In [5]:
def Sigmoid(value):
    return 1 / (1 + ag.Exp(-value))

In [6]:
x = ag.Node(1)
sig = Sigmoid(x)
sig.backward()

In [7]:
x.grad

0.19661193324148188

In [8]:
from math import exp
test = 1 / (1 + exp(-1))
print(test * (1 - test))

0.19661193324148185
